plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveModule.java

```diff
@@ -102,6 +104,10 @@ public void configure(Binder binder)
         binder.bind(FileFormatDataSourceStats.class).in(Scopes.SINGLETON);
         newExporter(binder).export(FileFormatDataSourceStats.class).withGeneratedName();
 
+        binder.bind(TrinoFileSystemCacheStats.class).toInstance(TrinoFileSystemCache.getFileSystemCacheStats());
+        newExporter(binder).export(TrinoFileSystemCacheStats.class)
+                .as(generator -> generator.generatedNameOf(TrinoFileSystemCache.class));
```

Comment:
nit: is there any particular reason to use `.as(generator -> generator.generatedNameOf(TrinoFileSystemCache.class));`  instead of `.withGeneratedName()` as in other places in this module ?

Reply:
i guess this is because we expose `TrinoFileSystemCacheStats` under name based on `TrinoFileSystemCache` (different classes)